mtp_api/apps/security/serializers.py

```diff
@@ -349,3 +352,76 @@ def reject(self, by):
             raise ValidationError(
                 detail=e.message_dict,
             )
+
+
+class CheckAutoAcceptRuleStateSerializer(serializers.ModelSerializer):
+    active = serializers.BooleanField(required=False)
+    added_by = BasicUserSerializer(read_only=True)
+
+    class Meta:
+        model = CheckAutoAcceptRuleState
+        fields = [
+            'added_by',
+            'active',
+            'reason',
+            'created'
+        ]
+        read_only_fields = (
+            'id',
+        )
+
+
+class CheckAutoAcceptRuleSerializer(serializers.ModelSerializer):
+    states = CheckAutoAcceptRuleStateSerializer(many=True, required=True)
+
+    def validate(self, attrs):
+        if len(attrs['states']) != 1:
+            raise ValidationError(
+                f'When creating or updating an auto-accept rule, states must be of length 1, not {len(attrs.states)}'
+            )
+        return super().validate(attrs)
+
+    def create(self, validated_data):
+        auto_accept_rule = CheckAutoAcceptRule.objects.create(
+            debit_card_sender_details=validated_data['debit_card_sender_details'],
+            prisoner_profile=validated_data['prisoner_profile'],
+        )
+        CheckAutoAcceptRuleStateSerializer().create(
+            validated_data={
+                'active': True,
```

Comment:
Oh interesting, the create request may have `"active": false` but we still create the first state with `active=True`. 

Do you reckon is worth adding a validation if the `active` field is passed as `False` on create?

Reply:
It's an internal API which is only used by a single client, I think it's fine for the API to create the initial `AutoAcceptRule` and `AutoAcceptRuleState` in the proper state without using the value for active given in the request. There's no reason to create an `AutoAcceptRule` in the `active=False` state, and if for some reason the POST endpoint was called for an existing check, then a database integrity error would be returned because of the unique constraint (`debit_card_details_id` and `prisoner_profile_id`) defined on the model.